ads-cert-examples/scala/src/main/scala/Main.scala

```diff
@@ -0,0 +1,87 @@
+import java.io.{FileInputStream, InputStreamReader}
+import java.nio.charset.StandardCharsets
+import java.security.spec.X509EncodedKeySpec
+import java.security._
+
+import org.apache.commons.codec.binary.Base64
+
+import org.bouncycastle.asn1.ASN1ObjectIdentifier
+import org.bouncycastle.asn1.x9.ECNamedCurveTable
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter
+import org.bouncycastle.openssl.{PEMKeyPair, PEMParser}
+
+import scala.util.{Failure, Success, Try}
+
+
+object Main {
+
+  private var privateKey: PrivateKey = _
```

Comment:
Using `var` is generally considered bad practice.  Should be able to have `.loadKeys` return a `Try[(PrivateKey, PublicKey)]`, or split that into a `.loadPrivateKey` that returns a `Try[PrivateKey]` and a `.loadPublicKey` that returns a `Try[PublicKey]`, though that would require splitting out the args in advance.

Since that's doing I/O to load the keys, it would not be amiss to use a `Future[_]` instead of a `Try[_]` which would also capture exceptions.